### PR TITLE
Lens

### PR DESCRIPTION
**Pull request by mfine**
What this library needs is more lens.

/cc @mookerji

### CODE REVIEW

**Comment by mookerji** [comment]
Oh man, another rev of the library, another >1k code review.

File: generator/sbpg/targets/resources/SbpTemplate.hs [hunk 1]
```diff
@@ -121,25 +122,23 @@ instance Binary SBPMsg where
...
+        | msgType == (((m | to_global))) = SBP(((m))) <$> parseJSON obj <*> parseJSON obj
+((*- endfor *))
+        | otherwise = SBPMsgUnknown <$> parseJSON obj
+  parseJSON _ = mzero
 
 merge :: Value -> Value -> Value
-merge (Object one) (Object two) = Object $ one <> two
-merge _ _ = undefined
+merge (Object one) (Object two) = Object (one <> two)
+merge _ _ = Null
```
Comment (mookerji) [inline generator/sbpg/targets/resources/SbpTemplate.hs]: Hmmmm, the problem here is that if we ever have something that isn't an `Object` map, it'll fail silently: https://hackage.haskell.org/package/aeson-0.9.0.1/docs/Data-Aeson.html#g:9 . I don't have strong opinions either way.

**Comment by mookerji** [comment]
lgtm

File: generator/sbpg/targets/resources/SbpTemplate.hs (hunk 1 above)
Comment (mfine) [inline generator/sbpg/targets/resources/SbpTemplate.hs]: Hmm, yeah - was wondering about that.... maybe something like this???

```haskell
merge :: Value -> Value -> Value
merge (Object one) (Object two) = Object (one <> two)
merge obj@(Object _) _ = obj
merge _ obj@(Object _) = obj
merge obj Null = obj
merge _ obj = obj
```

Definitely think we should yank the `undefined` either way.

Comment (mookerji) [inline generator/sbpg/targets/resources/SbpTemplate.hs]: Yes and yes.

Comment (mookerji) [inline generator/sbpg/targets/resources/SbpTemplate.hs]: Ideally, this would be an Either, but I'm happy not complicating the generated dispatch code.

**Pull request merged**